src/ring_theory/dedekind_domain/adic_valuation.lean

```diff
@@ -222,4 +240,118 @@ begin
   exact nat.eq_of_le_of_lt_succ mem nmem,
 end
 
+/-! ### Adic valuations on the field of fractions `K` -/
+/-- The `v`-adic valuation of `x ∈ K` is the valuation of `r` divided by the valuation of `s`,
+where `r` and `s` are chosen so that `x = r/s`. -/
+def valuation  (v : height_one_spectrum R) : valuation K (with_zero (multiplicative ℤ)) :=
+valuation.extend_to_localization v.int_valuation
+  (λ r hr, set.mem_compl (v.int_valuation_ne_zero' ⟨r, hr⟩)) K
+
+lemma valuation_def (x : K) : v.valuation x = valuation.extend_to_localization v.int_valuation
+  (λ r hr, set.mem_compl (v.int_valuation_ne_zero' ⟨r, hr⟩)) K  x :=
+rfl
+
+/-- The `v`-adic valuation of `r/s ∈ K` is the valuation of `r` divided by the valuation of `s`. -/
+lemma valuation_of_mk' {r : R} {s : non_zero_divisors R} :
+  v.valuation (is_localization.mk' K r s) = (v.int_valuation_def r)/(v.int_valuation_def s) :=
```

Comment:
Is there a reason you decided to use `v.int_valuation_def r` as opposed to `v.int_valuation r`? (similarly for `s`)
It doesn't look like `v.int_valuation r` (d)simplifies to `v.int_valuation_def r`.
Similarly for `valuation_of_algebra_map`.